Makefile

```diff
@@ -12,7 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: gen clean certs build
+ifndef REGISTRY
+$(error REGISTRY is not defined)
+endif
+ifndef PROJECT_ID
+$(error PROJECT_ID is not defined)
+endif
```

Comment:
Can you embed these checks into the targets that require them? The way this is written you'll have to specify these even if you have no need for them.

Reply:
Good point. Fixed.